app/src/common/shared/com/igalia/wolvic/downloads/DownloadsManager.java

```diff
@@ -230,18 +232,99 @@ public void onReceive(Context context, Intent intent) {
                 if (c != null) {
                     if (c.moveToFirst()) {
                         notifyDownloadsUpdate();
-                        notifyDownloadCompleted(Download.from(c));
+                        Download download = Download.from(c);
+                        notifyDownloadCompleted(download);
+
+                        // Copy the downloaded file to the public Downloads folder.
+                        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+                            copyToDownloadsFolder(download);
+                        }
                     }
                     c.close();
                 }
             }
         }
     };
 
+    @RequiresApi(api = Build.VERSION_CODES.Q)
+    private void copyToDownloadsFolder(Download download) {
+        Log.e(LOGTAG, "************************ copyToDownloadsFolder ************************");
+        Log.e(LOGTAG, "DISPLAY_NAME : " + download.getFilename());
+        Log.e(LOGTAG, "EXTERNAL_CONTENT_URI : " + MediaStore.Downloads.EXTERNAL_CONTENT_URI);
+        Log.e(LOGTAG, "SIZE : " + download.getSizeBytes());
+        Log.e(LOGTAG, "OutputFilePath : " + download.getOutputFilePath());
```

Comment:
Use `Log.d` (debug) for all those statements, `Log.e` is for errors.

Also perhaps put everythin in a single `Log.d` statement. Perhaps something like
`Log.d(LOGTAG, "Copying " + download.getFilename() + " to " + download.getOutputFilePath())`

It's better to have these things in a single line because that way it's easier to filter and get all the required information.